src/AuthCredentials.js

```diff
@@ -0,0 +1,82 @@
+import Credentials from './Credentials.js'
+import Storage from './Storage.js'
+import crypto from 'crypto'
+import { createJWT } from './JWT'
+
+// Generate random 18 byte hex string
+const random  = () => {
+  return new Promise((resolve, reject) => {
+    crypto.randomBytes(16, (err, buf) => {
+      if (err) reject(error)
+      resolve(buf.toString('hex'))
+    })
+  })
+}
+
+class AuthCredentials extends Credentials {
+  constructor (settings = {}) {
+    super(settings)
+    this.challengeStorage = settings.challengeStorage || new Storage('challenge')
+    this.responseStorage = settings.responseStorage || new Storage('response')
+    this.random = settings.random || random
+  }
+
+  // Create request token
+  createRequest (params = {}) {
+    const payload = {}
+    if (params.requested) {
+      payload.requested = params.requested
+    }
+    if (params.verified) {
+      payload.verified = params.verified
+    }
+    if (params.notifications) {
+      payload.permissions = ['notifications']
+    }
+    if (params.callbackUrl) {
+      payload.callback = params.callbackUrl
+    }
+
+    let pairId, challenge
+
+    return Promise.all([this.random(), this.random()]).then(rand => {
+      pairId = rand[0]
+      challenge = rand[1]
+      payload.challenge = `${pairId}.${challenge}`
```

Comment:
yeah I agree with moving challenge/auth code into a separate class, definitely possible and useful. I think I would still use a AuthCredentials class though, rather than adding more if lines based on configs, auth is a clear use case that is likely better defined by having its own class, it may also expand, and be usefully defined in some cases/context as providing auth alone.

was also looking to consider this another PR/story since this one has gotten so large